RAMLOAD builds with mcuboot are broken

### Describe the bug

Since #85254 was merged, `imgtool.py` expects `write-block-size` of nor device to be set and otherwise an error is thrown
```
Error: Invalid value for '--align': '--load-addr' is not one of '1', '2', '4', '8', '16', '32'.
```

However there is no fallback on this, so all related builds are broken and there is no consensus neither on how to set it (Kconfig vs dts, see [fix attempt here](https://github.com/zephyrproject-rtos/zephyr/pull/95152)).

### Regression

- [ ] This is a regression.

### Steps to reproduce

`west build -p always -b stm32n6570_dk/stm32n657xx --sysbuild -- -DSB_CONFIG_MCUBOOT_MODE_RAM_LOAD=y`
results in
```
Generating files from ~/zephyrproject/zephyr/samples/userspace/hello_world_user/build/stm32n6570_dk/stm32n657xx/hello_world_user/zephyr/zephyr.elf for board: stm32n6570_dk
Usage: imgtool.py sign [OPTIONS] INFILE OUTFILE
Try 'imgtool.py sign -h' for help.

Error: Invalid value for '--align': '--load-addr' is not one of '1', '2', '4', '8', '16', '32'.
ninja: build stopped: subcommand failed.
```

### Relevant log output

```shell

```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

_No response_

### Additional Context

_No response_

--- a/cmake/mcuboot.cmake
+++ b/cmake/mcuboot.cmake
@@ -142,7 +142,11 @@ function(zephyr_mcuboot_tasks)
     dt_reg_addr(slot1_partition_address PATH ${slot1_partition})
 
     dt_prop(write_block_size PATH "${flash_node}" PROPERTY "write-block-size")
-    set(imgtool_args --align ${write_block_size} --load-addr ${chosen_ram_address} ${imgtool_args})
+    if(CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD)
+      set(imgtool_args --align 1 --load-addr ${chosen_ram_address} ${imgtool_args})
+    else() # CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD_WITH_REVERT
+      set(imgtool_args --align ${write_block_size} --load-addr ${chosen_ram_address} ${imgtool_args})
+    endif()
     set(imgtool_args_alt_slot ${imgtool_args} --hex-addr ${slot1_partition_address})
     set(imgtool_args ${imgtool_args} --hex-addr ${slot0_partition_address})
   elseif(CONFIG_MCUBOOT_BOOTLOADER_MODE_SINGLE_APP_RAM_LOAD)

